### PR TITLE
Update GH/RDD Page automation to include I&T reports

### PR DESCRIPTION
**Pull request by c-suh**
## 🗒️ Summary

- Checks build release number
- If release 16+ and not current release, add links for I&T reports
- If release 16+ and current release, check for time window (i.e., June, December - build release months) or if flag is enabled to add links for I&T reports

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 95%

## ♻️ Related Issues

https://github.com/NASA-PDS/nasa-pds.github.io/issues/85

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: src/lasso/reports/gh_pages/summary.py [hunk 1]
```diff
@@ -11,6 +12,50 @@
...
+    2. If ADD_INT_REPORTS environment variable is set
+
+    :return: True if reports should be added, False otherwise
+    """
+    current_month = datetime.now().month
+    in_release_window = current_month in RELEASE_MONTHS
+
+    env_flag = os.environ.get("ADD_INT_REPORTS", "").lower() in ["true", "1", "yes"]
+
+    return in_release_window or env_flag
```
Comment (nutjob4life) [inline src/lasso/reports/gh_pages/summary.py]: In June or December, current build pages will always link to DDR/TestRail PDFs whether the PDFs exist yet or not. If the purpose is to avoid broken links until release artifacts are published, this should probably require the env flag, or check file existence.

File: src/lasso/reports/gh_pages/summary.py [hunk 2]
```diff
@@ -173,24 +218,52 @@ def write_md_file(herd, output_file_name, version):
...
+    # Add I&T reports section for version 16+ (before category links)
+    version_num = parse_version_number(version)
+    if version_num and version_num >= MIN_VERSION_FOR_INT_REPORTS:
+        should_show_reports = False
+
+        if is_current_build:
+            # For current build: check time window or environment flag
+            should_show_reports = should_add_int_reports_to_current_build()
+            if should_show_reports:
+                logger.info(f"Adding I&T reports to current build {version}")
```
Comment (nutjob4life) [inline src/lasso/reports/gh_pages/summary.py]: Logging calls should use `%` placeholders instead of f-strings (which are interpolated no matter if the logging is on or off):
```python
    logger.info("Adding I&T reports to current build %s", version)
```

File: src/lasso/reports/gh_pages/build_summaries.py [hunk 3]
```diff
@@ -20,6 +21,46 @@
...
+    Extracts the release version from lines like:
+        release = 'B17'
+        release = "B17.0"
+        release = '17'
+
+    :param conf_path: Path to conf.py file
+    :return: Current release version string (e.g., "17" from "B17"), or None if not found
+    """
+    try:
+        with open(conf_path, 'r') as f:
```
Comment (nutjob4life) [inline src/lasso/reports/gh_pages/build_summaries.py]: Should open anything non-binary with an explicit encoding:
```python
    with open(conf_path, 'r', encoding='utf-8') as f:
```

File: src/lasso/reports/gh_pages/build_summaries.py [hunk 4]
```diff
@@ -20,6 +21,46 @@
...
+            content = f.read()
+            # Look for: release = 'B17' or release = "B17.0" etc.
+            # Capture everything after = until the closing quote
+            match = re.search(r"release\s*=\s*['\"]([^'\"]+)['\"]", content)
+            if match:
+                release_string = match.group(1)
+                _logger.info(f"Found release in conf.py: {release_string}")
```
Comment (nutjob4life) [inline src/lasso/reports/gh_pages/build_summaries.py]: Logging should use `%` placeholders (see other comment; also applies to other logging calls).

**Review by nutjob4life — CHANGES_REQUESTED** [review]
Looks good; just a little "seasonal concern" with time of run versus time of what artifacts get produced. Also see a couple comments about logging calls with `%` placehoders and the `encoding` for `open`, otherwise super close!

File: src/lasso/reports/gh_pages/build_summaries.py (hunk 4 above)
Comment (c-suh) [reply src/lasso/reports/gh_pages/build_summaries.py]: Addressed in https://github.com/NASA-PDS/lasso-reports/pull/7/changes/5767318cb7b426801907492d83cc6d5abebeddc7.

File: src/lasso/reports/gh_pages/build_summaries.py (hunk 3 above)
Comment (c-suh) [reply src/lasso/reports/gh_pages/build_summaries.py]: Addressed in https://github.com/NASA-PDS/lasso-reports/pull/7/changes/5767318cb7b426801907492d83cc6d5abebeddc7.

File: src/lasso/reports/gh_pages/summary.py (hunk 2 above)
Comment (c-suh) [reply src/lasso/reports/gh_pages/summary.py]: Addressed in https://github.com/NASA-PDS/lasso-reports/pull/7/changes/5767318cb7b426801907492d83cc6d5abebeddc7.

File: src/lasso/reports/gh_pages/build_summaries.py (hunk 4 above)
Comment (c-suh) [reply src/lasso/reports/gh_pages/build_summaries.py]: Nope, found instances of this in summary.py as well. Addressing those now.

Comment (c-suh) [reply src/lasso/reports/gh_pages/build_summaries.py]: https://github.com/NASA-PDS/lasso-reports/pull/7/changes/269dde71f2e687ff3a1aa87852ff27c8c2668648

File: src/lasso/reports/gh_pages/summary.py (hunk 1 above)
Comment (c-suh) [reply src/lasso/reports/gh_pages/summary.py]: @nutjob4life a much better solution than the time-based check, which I was cringing about; thank you for the suggestion!

Comment (nutjob4life) [reply src/lasso/reports/gh_pages/summary.py]: Happy to help @c-suh ☺️

**Review by nutjob4life — APPROVED** [review]
Code deltas reviewed: ✓
Visual inspection: ✓
Testing: skipped
Approval: ✅
Thanks @c-suh!
